### PR TITLE
add optional chaining support

### PR DESCRIPTION
**Pull request by pmcclory-contrast**

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -1326,7 +1326,43 @@ export class Parser {
             } else if (this.lookahead.type === Token.Template && this.lookahead.head) {
                 const quasi = this.parseTemplateLiteral();
                 expr = this.finalize(this.startNode(startToken), new Node.TaggedTemplateExpression(expr, quasi));
-
+            } else if (this.match('?.')) {
+                this.nextToken();
+                if (this.match('(')) {
+                    var asyncArrow = maybeAsync && (startToken.lineNumber === this.lookahead.lineNumber);
```
Comment (pmcclory-contrast) [inline src/parser.ts:1332]: each of these conditions is pretty much copy and pasted from the respective parsing of the non-optional chaining versions of these.

The exception is the static member expression case, we can't use the `parseIdentifierName` function b/c it expects the token to still be at the `.` and handles advancing to the identifier, but we had to advance past to see which type of optional chaining this is.

File: src/nodes.ts [hunk 2]
```diff
@@ -301,6 +305,19 @@ export class ConditionalExpression {
...
+export class NullishConditionalExpression {
+    readonly type: string;
+    readonly test: Expression;
+    readonly alternate: Expression;
+    readonly nullish: boolean;
+    constructor(test: Expression, alternate: Expression) {
+        this.type = Syntax.ConditionalExpression;
+        this.test = test;
+        this.alternate = alternate;
+	this.nullish = true;
```
Comment (meandthemoon) [inline src/nodes.ts]: :trollface:

**Review by meandthemoon — APPROVED** [review]
These are usually a bit tricky and technical so great job getting this done quickly to support Node 14.

As mentioned in DM, there's some mystery still but reading through the changes makes sense and unit tests in the agent are passing so I'm good with this. Also, just to call out, we know these unit tests are failing here but also in `master`, but the fact that we're switching away from this homegrown solution makes me a little less concerned.

File: src/nodes.ts (hunk 2 above)
Comment (pmcclory-contrast) [reply src/nodes.ts]: yikes.  fixing.

**Pull request merged**